pkg/web/src/app/Providers/components/AddEditProviderModal/AddEditProviderModal.tsx

```diff
@@ -301,44 +330,105 @@ export const AddEditProviderModal: React.FunctionComponent<IAddEditProviderModal
+                                    {certificateQuery.data?.fingerprint}
+                                    {providerBeingEdited &&
+                                    fields.fingerprint?.cleanValue !==
+                                      certificateQuery.data.fingerprint ? (
+                                      <Alert
+                                        variant="warning"
+                                        isInline
+                                        title="The authenticity of this host cannot be established"
+                                      >
+                                        This certificate&apos;s fingerprint does not match the
+                                        previously known certificate. Manually validate the
+                                        fingerprint before proceeding.
+                                      </Alert>
+                                    ) : null}
+                                  </DescriptionListDescription>
+                                </DescriptionListGroup>
+                                <DescriptionListGroup>
+                                  <DescriptionListTerm>Expiration date</DescriptionListTerm>
+                                  <DescriptionListDescription id="expriry">
+                                    {certificateQuery.data?.valid_to}
+                                  </DescriptionListDescription>
+                                </DescriptionListGroup>
+                              </DescriptionList>
+                            </PanelMainBody>
+                          </PanelMain>
+                        </Panel>
+                        <Checkbox
+                          label="I trust the authenticity of this certificate"
+                          aria-label="Validate Certificate"
+                          id="certificate-check"
+                          name="certificateCheck"
+                          isChecked={fields?.isCertificateValid?.value}
+                          onChange={() => {
+                            if (fields.isCertificateValid?.value !== true) {
+                              if (
+                                fields?.fingerprint &&
+                                certificateQuery.data &&
+                                certificateQuery.data.fingerprint !== ''
+                              ) {
+                                fields.fingerprint?.setValue(certificateQuery.data.fingerprint);
+                              }
+                            } else {
+                              fields.fingerprint?.setValue('');
+                            }
+                            fields.isCertificateValid?.setValue(!fields.isCertificateValid.value);
+                          }}
+                        />
+                      </>
+                    ) : (
+                      <Text>Error: Cannot retrieve certificate</Text>
```

Comment:
Might be good to use an [`<Alert />`](https://www.patternfly.org/v4/components/alert/#static-live-region-alert) here.